File: C/unit1/recursive/tower_of_hanoi.c
#include <stdio.h>

void tower(int n, char A, char B, char C)
{
    if (n == 1)
    {
        printf("move disk 1 from %c to %c\n", A, B);
        return;
    }
    else
    {
        tower(n - 1, A, C, B);
        printf("move disk %d from %c to %c\n", n, A, B);
        tower(n - 1, C, B, A);
    }
}
int main()
{
    int n = 4;
    tower(n, 'A', 'C', 'B');
}

File: Java/unit1/recursive/tower_of_hanoi.java
package Java.unit1.recursive;

public class tower_of_hanoi {
    static void hanoi(int n, char form, char to, char aux) {
        if (n == 1) {
            System.out.println("move disk 1 from " + form + " to " + to);
            return;
        } else {
            hanoi(n - 1, form, aux, to);
            System.out.println("move disk " + n + " from " + form + " to " + to);
            hanoi(n - 1, aux, to, form);
        }
    }

    public static void main(String[] args) {
        int n = 4;
        hanoi(n, 'A', 'C', 'B');
    }
}
